### DISCUSSION TITLE
Why in my Azure function which uses PnP Core SDK , all the action/functions were done using SharePoint API

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1115] [Q&A] [answered]
I have developed an Azure Function using PnP Core SDK. and the Azure function integrate with Azure Active Directory App.

Now inside the Azure Active Directory App i defined the following under its API Permission;-
 
![image](https://user-images.githubusercontent.com/79726511/221272403-1998985b-dc51-45c3-b23b-fceb7d7d7ebd.png)

now i intentionally did not define any permission under Graph other than read user profile. and our Azure Function worked well, the azure function do the following main function/actions:-

1) Loop through a custom list items

2) Create folders and sub-folders

3) Define the Default for the folders as follow:-

![image](https://user-images.githubusercontent.com/79726511/221272858-d7064eea-d23f-4d46-af95-b795ce43dfbc.png)

now i thought that the Azure Function will not work, because as i know PnP core SDK use the Graph API first and it will only use the SharePoint API if the function is not available in the graph api. so can i say that in my case all the above 3 actions/ufnctions were done using SharePoint aPI? 

Second Question) and in the future if those functions become available inside Graph , then my Azure function will not work unless i grant further permissions to the Graph under the Azure AD App's API Permission?

Thanks

**Comment by jansenbe** [comment, marked as answer]
@mvcsharepointdev : PnP Core SDK requires both Graph and SharePoint permissions to work correctly. Please grant both.

**Reply by mvcsharepointdev** [reply]
@jansenbe  but as i mentioned my code is working well , without granting extra permissions to the Graph, why to grant extra unnecessary extra permission, am i correct?

**Reply by jansenbe** [reply]
Some operations use graph calls, your code works... Until you use a method that depends on Graph. So if today all works you can stick with that.

**Reply by mvcsharepointdev** [reply]
@jansenbe so this means that for all those operations:-

1.     Loop through a custom list items
2.     Create folders and sub-folders
3.     Define the Default for the folders.

PnP will use SharePoint REST API and not Graph ? am i correct?

**Reply by jansenbe** [reply]
Yes
